src/core/interfaces.ts

```diff
@@ -46,4 +46,12 @@ module Plottable {
     topLeft: Point;
     bottomRight: Point;
   };
+
+  export interface Entity<C extends Component> {
+    datum: any;
+    index: number;
+    position: Point;
+    selection: d3.Selection<any>;
+    component: C;
```

Comment:
Previously `index` would have been the position in the `Dataset` array, and `position` is a representative location on the screen. Possibly one or both of them doesn't belong in the base `Entity`?